common/types/block.go

```diff
@@ -105,6 +106,40 @@ func (id NodeID) ShortString() string {
 	return Shorten(name, 5)
 }
 
+// BytesToNodeID deserializes a byte slice into a NodeID
+func BytesToNodeID(b []byte) (*NodeID, error) {
+	if len(b) < 32 {
```

Comment:
Why do we allow a range of lengths (32-64 bytes) here, rather than requiring a fixed length?